jina/executors/indexers/cache.py

```diff
@@ -59,28 +69,24 @@ def close(self):
             pickle.dump(self.id_to_cache_val, open(self.path + '.ids', 'wb'))
             pickle.dump(self.cache_val_to_id, open(self.path + '.cache', 'wb'))
 
-    supported_fields = [ID_KEY, CONTENT_HASH_KEY]
-    default_field = ID_KEY
-
-    def __init__(self, index_filename: Optional[str] = None, field: Optional[str] = None, *args, **kwargs):
-        """Create a new DocCache.
+    default_fields = (ID_KEY,)
 
-        :param index_filename: file name for storing the cache data
-        :param field: field to cache on (ID_KEY or CONTENT_HASH_KEY)
-        """
+    def __init__(self, index_filename: Optional[str] = None, fields: Optional[Tuple[str]] = None, *args, **kwargs):
         if not index_filename:
             # create a new temp file if not exist
             index_filename = tempfile.NamedTemporaryFile(delete=False).name
         super().__init__(index_filename, *args, **kwargs)
-        self.field = field or self.default_field
-        if self.field not in self.supported_fields:
-            raise ValueError(f"Field '{self.field}' not in supported list of {self.supported_fields}")
+        fields = fields or self.default_fields
```

Comment:
we try to avoid assignments to the same variable.

Reply:
Done, thanks